src/comments/comment.service.ts

```diff
@@ -0,0 +1,360 @@
+    }
+
+    // 创建评论
+    const comment = new Comment();
+    comment.content = content;
+    comment.commentableType = 'question';
+    comment.commentableId = question.id;
+    const createdComment = await this.commentsRepository.save(comment);
+    const commentDto: CommentDto = {
+      id: createdComment.id,
+      content: createdComment.content,
+      commentableId: comment.commentableId,
+      commentableType: comment.commentableType,
+      user: comment.user,
+      quote: {
+        quote_id: null,
+        quote_user: null,
+      },
+      agree_count: 0,
+      disagree_count: 0,
+      agree_type: 0,
+      created_at: comment.createdAt.getDate(),
+    };
+    return commentDto;
+  }
+
+  async commentComment(
+    commentId: number,
+    content: string,
+  ): Promise<CommentDto> {
+    const Comments = await this.questionsRepository.findOneBy({
+      id: commentId,
+    });
+
+    if (!Comments) {
+      throw new Error('Answer not found');
+    }
+
+    const comment = new Comment();
+    comment.content = content;
+    comment.commentableType = 'question';
+    comment.commentableId = Comments.id;
+
+    const createdComment = await this.commentsRepository.save(comment);
+
+    const commentDto: CommentDto = {
+      id: createdComment.id,
+      content: createdComment.content,
+      commentableId: comment.commentableId,
+      commentableType: comment.commentableType,
+      user: comment.user,
+      quote: {
+        quote_id: null,
+        quote_user: null,
+      },
+      agree_count: 0,
+      disagree_count: 0,
+      agree_type: 0,
+      created_at: comment.createdAt.getDate(),
+    };
+
+    return commentDto;
+  }
+
+  async getCommentDetail(
+    answerId: number,
+    commentId: number,
+    pageStart: number = 0,
+    pageSize: number = 20,
+  ) {
+    const comment = await this.commentsRepository.findOneBy({
+      id: commentId,
+    });
+    if (!comment) {
+      throw new NotFoundException('Comment not found');
+    }
+    const subComments = await this.commentsRepository.find({
+      where: {
+        parentComment: { id: commentId },
+      },
+      order: {
+        id: 'ASC',
+      },
+      skip: pageStart,
+      take: pageSize,
+    });
+    return {
+      comment,
+      subComments,
+      page: {
+        pageStart,
+        pageSize,
+      },
+    };
+  }
+  async agreeComment(
+    id: number,
+    answerId: number,
+    commentId: number,
+    agreeType: number,
+  ) {
+    const comment = await this.commentsRepository.findOneBy({
+      id: commentId,
+    });
+    if (!comment) {
+      throw new NotFoundException(`Comment with id ${commentId} not found`);
+    }
+    switch (agreeType) {
+      case 0:
+        break;
+      case 1:
+        comment.agreecount = comment.agreecount + 1;
+        break;
+      case 2:
+        comment.disagreecount = comment.disagreecount + 1;
+        break;
+      default:
+        throw new Error('Invalid agreeType value');
```

Comment:
Unless you are sure that the error is a logical error in the internal code, implement the class you inherited from `BaseError`. Otherwise ErrorFilter will not work properly.